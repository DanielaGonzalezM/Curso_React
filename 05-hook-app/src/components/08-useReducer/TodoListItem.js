import React from "react";
import PropTypes from "prop-types";

const TodoListItem = ({ todo, index, handleDelete, handleToggle }) => {
  return (
    <li className="list-group-item" key={todo.id}>
      <p
        onClick={() => handleToggle(todo.id)}
        className={`${todo.done && "complete"}`}
      >
        {index + 1}. {todo.desc}
      </p>
      <button onClick={() => handleDelete(todo.id)} className="btn btn-danger">
        Borrar
      </button>
    </li>
  );
};

TodoListItem.propTypes = {
  todos: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToggle: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
};

TodoListItem.defaultProps = {
  todos: [],
  handleDelete: () => {},
  handleToggle: () => {},
  index: 0,
};

export default TodoListItem;
