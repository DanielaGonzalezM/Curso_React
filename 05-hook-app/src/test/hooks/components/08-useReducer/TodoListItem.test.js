import { shallow } from "enzyme";
import TodoListItem from "../../../../components/08-useReducer/TodoListItem";
import { demoTodos } from "../../../fixtures/demoTodos";

describe("Pruebas TodoListItem", () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      index={0}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );
  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de llamar la función handleDelete", () => {
    wrapper.find("button").simulate("click");
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test("debe de llamar la función handleToggle", () => {
    wrapper.find("p").simulate("click");
    expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test("debe mostrar el texto correcto", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(`1. ${demoTodos[0].desc}`);
  });

  test("debe tener la clase complete so el TODO.done=true", () => {
    const todo = demoTodos[0];
    todo.done = true;
    const wrapper = shallow(
      <TodoListItem
        todo={todo}
      />
    );
    console.log(wrapper.html());
    expect(wrapper.find("p").hasClass("complete")).toBe(true);
  });
});
