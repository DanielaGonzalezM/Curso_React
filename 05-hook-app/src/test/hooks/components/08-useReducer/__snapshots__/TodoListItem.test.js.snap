// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas TodoListItem debe de mostrarse correctamente 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="undefined"
    onClick={[Function]}
  >
    1
    . 
    Aprender React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
